### PR TITLE
generate_test: implement example testing

### PR DESCRIPTION
**Pull request by mikedanese**
fixes https://github.com/bazelbuild/rules_go/issues/1341

### CODE REVIEW

**Comment by mikedanese** [comment]
@jayconrod that should fix it. Is there anyway to test that a go_test fails?

**Comment by mikedanese** [comment]
Now the example with empty output runs, and the one without the output directive does not.
```console
$ bazel test --test_output=all --test_arg=-test.v //tests/core/go_test:example_test
INFO: Analysed target //tests/core/go_test:example_test (0 packages loaded).
INFO: Found 1 test target...
INFO: From Testing //tests/core/go_test:example_test:
==================== Test output for //tests/core/go_test:example_test:
=== RUN   ExampleEmptyOutputTest
--- PASS: ExampleEmptyOutputTest (0.00s)
=== RUN   ExampleHelloWorld
--- PASS: ExampleHelloWorld (0.00s)
PASS
================================================================================
Target //tests/core/go_test:example_test up-to-date:
  bazel-bin/tests/core/go_test/linux_amd64_stripped/example_test
INFO: Elapsed time: 1.103s, Critical Path: 0.12s
INFO: Build completed successfully, 2 total actions
//tests/core/go_test:example_test                                        PASSED in 0.1s

Executed 1 out of 1 test: 1 test passes.
```

File: tests/core/go_test/example_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	fmt.Println("expected: " + Expected)
+	fmt.Println("got: " + Got)
+	// Output:
+	// Hello Example!
+	// expected: Example
+	// got: Example
+}
+
+func ExampleDontTestMe() {
+	fmt.Println("Dont Test Me!")
```
Comment (jayconrod) [inline tests/core/go_test/example_test.go]: Maybe replace this with a `panic`. I'm sure that would be reported if it were run.

**Comment by jayconrod** [comment]
About testing whether a `go_test` fails, there is a way using `bazel_test` (in `//tests:bazel_tests.bzl`), but that's very expensive to run (involves running Bazel in a separate, fake workspace). Probably no need for that here. Panicking inside examples not intended to be run should be good enough.

**Comment by mikedanese** [comment]
@jayconrod fixed.

**Comment by ixdy** [comment]
@jayconrod is this all set to merge?

**Comment by jayconrod** [comment]
Yes, sorry, slipped off my radar.

**Pull request merged**